sysadmin/build.yml

```diff
@@ -0,0 +1,550 @@
+- name: opencontext setup
+  hosts: '{{vname}}'
+  vars:
+    user: "{{ ansible_user }}"
+    user_owner: www-data
+    group_owner: www-data
+    target: '{{vname}}'
+    migrate: "{{do_migrate | default('true')}}"
+    install_db: "{{install_db | default('true')}}"
+    install_solr_data: "{{install_solr_data | default('true')}}"
+
+    # postgres vars -- for LOCAL postgres installation
+    postgresql_python_library: python-psycopg2
+    postgresql_user: postgres
+    postgresql_group: postgres
+    postgresql_databases:
+      - name: "{{SECRET_KEYS.DATABASES_NAME}}"
+    postgresql_users:
+      - name: "{{SECRET_KEYS.DATABASES_USER}}"
+        password: "{{SECRET_KEYS.DATABASES_PASSWORD}}"
+
+    # using https://github.com/rdhyee/ansible-python
+    # forked from https://github.com/whiskerlabs/ansible-python
+    python_version: 3.7.4
+    python_bin: /usr/bin
+    python_pip_version: 19.1.1
+    python_upgrade_packaging_tools: false
+    # python_setuptools_version: 40.4.2
+    # python_virtualenv_version: 16.0.0
+
+    # git parameters
+    # expected to be overridden in extra.yml
+    git_user_name: ""
+    git_user_email: ""
+
+  become: yes
+
+  gather_facts: True
+  pre_tasks:
+
+    - name: Include vars of extra.yml
+      include_vars:
+        file: extra.yml
```

Comment:
I will document how to use this in practice.  e.g.,

```YML
oc_install_dir: /var/oc-venv

allowed_host: happy.opencontext.org
deployed_host: https://happy.opencontext.org
deployed_site_name: "happy OCserver"
server_name: happy.opencontext.org

git_user_name: "Raymond Yee"
git_user_email: "raymond.yee@gmail.com"
```

Reply:
One thing that would be nice to work out: how to override variables that are in a dict like `SECRET_KEYS`

Reply:
OK